resources/openshift/controller-with-template.yaml

```diff
@@ -478,3 +558,127 @@ objects:
+  displayName: Group ID
+  name: KAFKA_CONNECT_GROUP_ID
+  required: true
+  value: connect-cluster
+- description: Converter class used to convert between Kafka Connect format and the serialized form that is written to Kafka.
+  displayName: Key Converter
+  name: KAFKA_CONNECT_KEY_CONVERTER
+  required: true
+  value: org.apache.kafka.connect.json.JsonConverter
+- description: Set to false to use schemaless format
+  displayName: Enable schemas for key converters
+  name: KAFKA_CONNECT_KEY_CONVERTER_SCHEMAS_ENABLE
+  value: "true"
+- description: Converter class used to convert between Kafka Connect format and the serialized form that is written to Kafka.
+  displayName: Value Converter
+  name: KAFKA_CONNECT_VALUE_CONVERTER
+  required: true
+  value: org.apache.kafka.connect.json.JsonConverter
+- description: Set to false to use schemaless format
+  displayName: Enable schemas for value converters
+  name: KAFKA_CONNECT_VALUE_CONVERTER_SCHEMAS_ENABLE
+  value: "true"
+- description: Replication factor for config storage topic
+  displayName: Config replication factor
+  name: KAFKA_CONNECT_CONFIG_STORAGE_REPLICATION_FACTOR
+  value: "3"
+- description: Replication factor for offset storage topic
+  displayName: Offset replication factor
+  name: KAFKA_CONNECT_OFFSET_STORAGE_REPLICATION_FACTOR
+  value: "3"
+- description: Replication factor for status storage topic
+  displayName: Status replication factor
+  name: KAFKA_CONNECT_STATUS_STORAGE_REPLICATION_FACTOR
+  value: "3"
+- description: Image repository name
+  displayName: Repository Name
+  name: IMAGE_REPO_NAME
+  value: strimzi
+- description: Image name
+  displayName: Image Name
+  name: IMAGE_NAME
+  value: kafka-connect
+- description: Image tag
+  displayName: Image tag
+  name: IMAGE_TAG
+  value: latest
+- description: Image repository name of the S2I Docker image
+  displayName: S2I image repository
+  name: S2I_IMAGE_REPO_NAME
+  value: strimzi
+- description: Name of the S2I Docker image
+  displayName: S2I image name
+  name: S2I_IMAGE_NAME
+  value: kafka-connect-s2i
+- description: Tag of the S2I Docker image
+  displayName: S2I image tag
+  name: S2I_IMAGE_TAG
+  value: latest
+- description: Number of seconds after the container has started before healthcheck probes are initiated.
+  displayName: Kafka Connect healthcheck initial delay
+  name: HEALTHCHECK_DELAY
+  value: "60"
+- description: Number of seconds after which the probe times out.
+  displayName: Kafka Connect healthcheck timeout
+  name: HEALTHCHECK_TIMEOUT
+  value: "5"
+objects:
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: ${CLUSTER_NAME}
+    labels:
+      strimzi.io/kind: cluster
+      strimzi.io/type: kafka-connect
+  data:
+    nodes: "${INSTANCES}"
+    image: "${IMAGE_REPO_NAME}/${IMAGE_NAME}:${IMAGE_TAG}"
+    s2i: |-
+      {
+        "enabled": true,
```

Comment:
Maybe it's possible doing something like for metrics ? If `s2i` field is specified then it's enabled and the only JSON field will be `sourceImage`. In order to disable S2I you should just not define the `s2i` field. We could get rid of `enabled` field in this way. Or there is any specific reason I miss about using `enabled` explicitly ?

Reply:
Actually, the sourceImage is not mandatory - if not specified than the default value is used. and having only `s2i: { }` to enable S2I seemed strange. So this is why I added the `enabled` field. That would be completely unnecessary for Storage because there you have the type as mandatory option.

Reply:
Understood. If documented we could say that `{ }` just mean using default configuration but still avoiding the `s2i` field for disabling in order to have a consistent way for other potential future cases (like we already have with `storage`). Btw I'm fine to stick with current status ... it's just a nit :-)

Reply:
Yeah, `s2i: { }` would basically mean S2I disabled. It should go into the documentation (once we have any).